extensions/peerjs.js

```diff
@@ -0,0 +1,957 @@
+const menuIconURI =
+  "data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTAiIGhlaWdodD0iNTAiIHZpZXdCb3g9IjAgMCA1MCA1MCIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPGcgY2xpcC1wYXRoPSJ1cmwoI2NsaXAwXzE4OV8xNTApIj4KPHBhdGggZD0iTTAgMjVDMCAxMS4xOTI5IDExLjE5MjkgMCAyNSAwQzM4LjgwNzEgMCA1MCAxMS4xOTI5IDUwIDI1QzUwIDM4LjgwNzEgMzguODA3MSA1MCAyNSA1MEMxMS4xOTI5IDUwIDAgMzguODA3MSAwIDI1WiIgZmlsbD0iI0U5NjE1MSIvPgo8cGF0aCBkPSJNMjcgMzNIMjNWNDdIMjdWMzNaIiBmaWxsPSJ3aGl0ZSIgc3Ryb2tlPSJibGFjayIgc3Ryb2tlLXdpZHRoPSIwLjUiLz4KPHBhdGggZD0iTTIxLjM2MzYgMzJMMTcgMjMuNVYxNUgyMC4zOTM5TDIzLjMwMyAxOC4zMDU2SDI2LjIxMjFMMjkuMTIxMiAxNUgzM1YyMy41TDI4LjE1MTUgMzJIMjEuMzYzNloiIGZpbGw9IndoaXRlIiBzdHJva2U9ImJsYWNrIiBzdHJva2Utd2lkdGg9IjAuNSIvPgo8cGF0aCBkPSJNMTggM1YxMy41SDIyLjVWOS41SDIxLjVWNkgxOVYzSDE4WiIgZmlsbD0id2hpdGUiIHN0cm9rZT0iYmxhY2siIHN0cm9rZS13aWR0aD0iMC41Ii8+CjxwYXRoIGQ9Ik0zMiAzVjEzLjVIMjcuNVY5LjVIMjguNVY2SDMxVjNIMzJaIiBmaWxsPSJ3aGl0ZSIgc3Ryb2tlPSJibGFjayIgc3Ryb2tlLXdpZHRoPSIwLjUiLz4KPHBhdGggZD0iTTIzLjUgMTYuNUgyNi41VjguNUgyNy41VjZIMjIuNVY4LjVIMjMuNVYxNi41WiIgZmlsbD0id2hpdGUiIHN0cm9rZT0iYmxhY2siIHN0cm9rZS13aWR0aD0iMC41Ii8+CjxwYXRoIGQ9Ik0yMSA1VjNIMjJWNUgyMVoiIGZpbGw9IndoaXRlIiBzdHJva2U9ImJsYWNrIiBzdHJva2Utd2lkdGg9IjAuNSIvPgo8cGF0aCBkPSJNMjMgNVYzSDI0VjVIMjNaIiBmaWxsPSJ3aGl0ZSIgc3Ryb2tlPSJibGFjayIgc3Ryb2tlLXdpZHRoPSIwLjUiLz4KPHBhdGggZD0iTTI2IDVWM0gyN1Y1SDI2WiIgZmlsbD0id2hpdGUiIHN0cm9rZT0iYmxhY2siIHN0cm9rZS13aWR0aD0iMC41Ii8+CjxwYXRoIGQ9Ik0yOCA1VjNIMjlWNUgyOFoiIGZpbGw9IndoaXRlIiBzdHJva2U9ImJsYWNrIiBzdHJva2Utd2lkdGg9IjAuNSIvPgo8L2c+CjxkZWZzPgo8Y2xpcFBhdGggaWQ9ImNsaXAwXzE4OV8xNTAiPgo8cmVjdCB3aWR0aD0iNTAiIGhlaWdodD0iNTAiIGZpbGw9IndoaXRlIi8+CjwvY2xpcFBhdGg+CjwvZGVmcz4KPC9zdmc+Cg==";
+
+	
+    function fatalAlert(message) {
+        alert(message);
+        throw new Error(message);
+    }
+
+    // Require the extension to be unsandboxed
+    if (!Scratch2.extensions.unsandboxed) {
+        fatalAlert("The PeerJS extension must be loaded in an unsandboxed environment.");
+    }
+
+    // Require access to the VM and/or runtime
+    if (!Scratch2.vm || !Scratch2.vm.runtime) {
```

Comment:
If this extension IS unsandboxed, then it should always have access to the `vm` and `runtime` iirc, but I don't know what weird method you used.